src/node/internal/process.ts

```diff
@@ -20,13 +20,107 @@ export function nextTick(cb: Function, ...args: unknown[]) {
   });
 }
 
-// Note that there is no process-level environment in workers so the process.env
-// object is initially empty. This is different from Node.js where process.env
-// picks up values from the operating system environment. The configured bindings
-// for the worker are accessible from the env argument passed into the fetch
-// handler and have no impact here.
+// Decide if a value can round-trip to JSON without losing any information.
+function isJsonSerializable(
+  value: any,
+  seen: Set<Object> = new Set()
+): boolean {
+  switch (typeof value) {
+    case 'boolean':
+    case 'number':
+    case 'string':
+      return true;
 
-export const env = new Proxy(utilImpl.getEnvObject(), {
+    case 'object': {
+      if (value === null) {
+        return true;
+      }
+
+      if (seen.has(value)) {
+        // Don't allow cycles or aliases. (Non-cyclic aliases technically could be OK, but a
+        // round trip to JSON would lose the fact that they are aliases.)
+        return false;
+      }
+      seen.add(value);
+
+      if ('toJSON' in value) {
```

Comment:
```suggestion
      if (typeof value.toJSON === 'function') {
```

Just to prevent against silly user code bugs (I've actually seen things like `foo.toJSON = {}` in the wild.... yes, I know, it's awful)